src/formatDate.js

```diff
@@ -9,6 +9,59 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const separatorOld = fromFormat[fromFormat.length - 1];
+  const separatorNew = toFormat[toFormat.length - 1];
+
+  const fromFormatSlice = fromFormat.slice(0, 3);
+  const dateSplit = date.split(separatorOld);
+  const result = [];
+
+  const dateObject = {};
+
+  for (const type in fromFormatSlice) {
```

Comment:
Using `for...in` to iterate over arrays is not recommended as it iterates over enumerable properties, which can lead to unexpected behavior. Consider using `for...of` or a traditional `for` loop to iterate over `fromFormatSlice`.